var/spack/repos/builtin/packages/occa/package.py

```diff
@@ -33,91 +38,177 @@ class Occa(Package):
     version("0.2.0", tag="v0.2.0", commit="2eceaa5706ad6cf3a1b153c1f2a8a2fffa2d5945")
     version("0.1.0", tag="v0.1.0", commit="381e886886dc87823769c5f20d0ecb29dd117afa")
 
+    depends_on("hip", when="+rocm")
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
     depends_on("fortran", type="build")  # generated
+    depends_on("cmake@3.30:", type="build", when="build_system=cmake")
+    depends_on("gmake", type="build")
 
-    variant("cuda", default=True, description="Activates support for CUDA")
-    variant("openmp", default=True, description="Activates support for OpenMP")
-    variant("opencl", default=True, description="Activates support for OpenCL")
+    build_system(
+        conditional("cmake", when="@1.1.0:"),
+        conditional("makefile", when="@:1.0.9"),
+        default="makefile",
```

Comment:
This default is inconsistent with versions defaults. Can we default to `cmake`, or is there a reason for this?

Reply:
Older versions of OCCA (<=1.0.9) doesn't support `CMake.`

Reply:
This is not what you say in the directive

Reply:
Sorry, it is still not clear to me what the issue is. But seems like what you getting at is the following:
```python
build_system(conditional("cmake", when="@1.1.0:"), "makefile", default="cmake")
```

Reply:
What I am suggesting is:
```suggestion
        default="cmake",
```
I guess, whether you want to remove the condition on the `makefile` build-system is up to you, as a package maintainer.